src/mpid/ch4/netmod/ofi/ofi_am_impl.h

```diff
@@ -571,6 +567,8 @@ static inline int MPIDI_OFI_do_inject(int rank,
         goto fn_exit;
     }
 
+    MPIR_CHKLMEM_MALLOC(buff, char *, MPIDI_Global.max_buffered_send, mpi_errno, "buff", MPL_MEM_BUFFER);
```

Comment:
This was a clever fix, but I think this fits better with the overall MPICH style. I'm a little wary of variable arrays in general. In theory, the CHKLMEM macros can work for stack allocations and should be just as fast. The only question is, why do we default to `enable_alloca=no`?

Reply:
I don't think using `alloca` would solve this problem. From my background reading on this warning, it will pop up for `alloca` just as much as any other stack allocation.

Reply:
Can we confirm? If you add `--enable-alloca` to configure it should enable usage for all `CHKLMEM` macros.

Reply:
Eh, alloca seems like a really bad idea after reading some more. Maybe I should drop that and just consider if malloc-based CHKLMEM is OK or not here.

Reply:
How about we assert at init time `prov_use->tx_attr->inject_size < UINT16_MAX` and convert the type of `MPIDI_Global.max_buffered_send`?

Reply:
Good idea. We probably don't even have to assert. We can simply cap `inject_size` at `UINT16_MAX` at init time.

Reply:
I mean, `MPIDI_Global.max_buffered_send = MPL_MIN(prov_use->tx_attr->inject_size, UINT16_MAX);`

Reply:
That works for me. Can one of you guys make that change, force push and restart the tests?

Reply:
Done. I redid the entire final patch.